tests/Common/TokensTest.php

```diff
@@ -1045,4 +1045,37 @@ public function testTokenWithoutTokensManagePermissionCanListOnlyOwnTokenEvents(
             $this->assertEquals(403, $e->getCode());
         }
     }
+
+    public function testGuestTokenCreateRestrictedToken()
+    {
+        $client = new \Keboola\StorageApi\Client(array(
+            'token' => STORAGE_API_GUEST_TOKEN,
+            'url' => STORAGE_API_URL,
+            'backoffMaxTries' => 1,
+            'jobPollRetryDelay' => function () {
+                return 1;
+            },
+        ));
+
+        $token = $client->verifyToken();
+
+        $this->assertTrue($token['isMasterToken']);
+        $this->assertFalse($token['canManageTokens']);
+
+        $options = new TokenCreateOptions();
+        $options->setDescription('Autosave test');
+        $options->setExpiresIn(60 * 5);
+
+        $tokenId = $client->createToken($options);
+
+        $token = $this->_client->getToken($tokenId);
+
+        $this->assertFalse($token['isMasterToken']);
+        $this->assertFalse($token['canManageBuckets']);
```

Comment:
Ja by som mozno uvazoval o tom, ze by som realne zavolal nejaku akciu na ktoru nemam mat pravo a otestoval ci mi to vrati chybu

Reply:
To bych řekl, že testujeme jinde, ne? (\Keboola\Test\Common\TokensTest::testTokenWithoutTokensManagePermissionCanListOnlyOwnTokenEvents)

Reply:
Ale je pravdam že to možná netestujeme dostatečně zevrubně....

Reply:
Především za mě chybí test  na to že tam reálně nemůže poslat třeba canManageToken nebo canReadAllFileUploads (to ideálně) nebo v současné konfiguraci aspoň, že když to tam pošle tak je to vyignoruje.

Reply:
Jo to je vlastne taky pravda.

Reply:
> To bych řekl, že testujeme jinde, ne? (\Keboola\Test\Common\TokensTest::testTokenWithoutTokensManagePermissionCanListOnlyOwnTokenEvents)

Ano to se testuje už někde jinde.

Test jsem upravil aby se tam posílaly i ty parametry co se mají vyignorovat.